android-client/Espoo/app/src/main/java/com/espoo/android/view/LoginActivity.kt

```diff
@@ -58,19 +58,23 @@ class LoginActivity : AppCompatActivity() {
         } else {
             service.login(AuthData(UserLogin(userName, password))).enqueue(object :Callback<User> {
                 override fun onResponse(call: Call<User>, response: Response<User>) {
-                    response.body()?.let {
-                        storeLoginData(it)
-                        openMainActivity()
-                    }
-                    response.headers()["Authorization"]?.let {
-                        sessionManager.storeData(API_TOKEN, it)
+                    if (response.isSuccessful) {
+                        response.body()?.let {
+                            storeLoginData(it)
+                            openMainActivity()
+                        }
+                        response.headers()["Authorization"]?.let {
+                            sessionManager.storeData(API_TOKEN, it)
+                        }
+                    } else{
+                        response.errorBody()?.let {
+                            val jsonResponse = JSONObject(it.string())
+                            Toast.makeText(applicationContext, jsonResponse.getString("error"), Toast.LENGTH_LONG)
+                                .show()
```

Comment:
I Belive that this will be implements in many places, is considerable implement a function to this?

Reply:
I moved the error response check to the API layer, removing it from the view layer.